service/frontend/workflow_handler.go

```diff
@@ -5180,6 +5202,569 @@ func (wh *WorkflowHandler) RespondNexusTaskFailed(ctx context.Context, request *
+				StartOperation: &nexuspb.StartOperationRequest{
+					Service:        request.Service,
+					Operation:      request.Operation,
+					RequestId:      request.RequestId,
+					Callback:       request.Callback,
+					Payload:        request.Payload,
+					CallbackHeader: request.CallbackHeader,
+					Links:          request.Links,
+				},
+			},
+		},
+	}
+
+	resp, err := wh.matchingClient.DispatchNexusTask(ctx, matchingReq)
+	if err != nil {
+		if common.IsContextDeadlineExceededErr(err) {
+			return &workflowservice.StartNexusOperationResponse{
+				Variant: &workflowservice.StartNexusOperationResponse_HandlerError{
+					HandlerError: errNexusUpstreamTimeout,
+				},
+			}, nil
+		}
+		return nil, err
+	}
+
+	switch t := resp.GetOutcome().(type) {
+	case *matchingservice.DispatchNexusTaskResponse_HandlerError:
+		return &workflowservice.StartNexusOperationResponse{
+			Variant: &workflowservice.StartNexusOperationResponse_HandlerError{
+				HandlerError: t.HandlerError,
+			},
+		}, nil
+	case *matchingservice.DispatchNexusTaskResponse_Response:
+		switch t := t.Response.GetStartOperation().GetVariant().(type) {
+		case *nexuspb.StartOperationResponse_SyncSuccess:
+			return &workflowservice.StartNexusOperationResponse{
+				Links: t.SyncSuccess.Links,
+				Variant: &workflowservice.StartNexusOperationResponse_SyncSuccess{
+					SyncSuccess: &workflowservice.StartNexusOperationResponse_Sync{
+						Result: t.SyncSuccess.Payload,
+					},
+				},
+			}, nil
+		case *nexuspb.StartOperationResponse_AsyncSuccess:
+			return &workflowservice.StartNexusOperationResponse{
+				Links: t.AsyncSuccess.Links,
+				Variant: &workflowservice.StartNexusOperationResponse_AsyncSuccess{
+					AsyncSuccess: &workflowservice.StartNexusOperationResponse_Async{
+						OperationToken: t.AsyncSuccess.OperationToken,
+					},
+				},
+			}, nil
+		case *nexuspb.StartOperationResponse_OperationError:
+			return &workflowservice.StartNexusOperationResponse{
+				Variant: &workflowservice.StartNexusOperationResponse_Unsuccessful_{
+					Unsuccessful: &workflowservice.StartNexusOperationResponse_Unsuccessful{
+						OperationError: t.OperationError,
+					},
+				},
+			}, nil
+		}
+	}
+
+	return &workflowservice.StartNexusOperationResponse{
+		Variant: &workflowservice.StartNexusOperationResponse_HandlerError{
+			HandlerError: errNexusUnexpectedOutcome,
+		},
+	}, nil
+}
+
+func (wh *WorkflowHandler) RequestCancelNexusOperation(ctx context.Context, request *workflowservice.RequestCancelNexusOperationRequest) (_ *workflowservice.RequestCancelNexusOperationResponse, retErr error) {
+	defer log.CapturePanic(wh.logger, &retErr)
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+	ns, err := wh.namespaceRegistry.GetNamespace(namespace.Name(request.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	switch t := request.GetTarget().GetVariant().(type) {
+	case *nexuspb.TaskDispatchTarget_Endpoint:
+		// forward via HTTP
```

Comment:
Not necessarily a forward.

```suggestion
		// dispatch via HTTP
```